screens/ImagesScreen/index.js

```diff
@@ -2,40 +2,48 @@ import PropTypes from 'prop-types';
 import {useState, useContext, useEffect} from 'react';
 import {Image, ScrollView, View} from 'react-native';
 import Header from '../../components/Header/Header';
+import ImagesModal from '../../components/Permission/children/images';
+import PermissionModal from '../../components/Permission/Modal';
 import {GlobalDataContext} from '../../contexts/context';
-import MyBottomSheet from './BottomSheet';
 import RenderImagePairs from './RenderImagePairs';
 import styles from './style';
 
 const ImagesScreen = ({navigation}) => {
   const [sheet, setSheet] = useState('');
   const {arrayImages, setArrayImage, setImageUrl} =
     useContext(GlobalDataContext);
-  const [isBottomSheetVisible, setBottomSheetVisible] = useState(false);
   const [photoId, setPhotoId] = useState(null);
+  const [isModalVisible, setModalVisible] = useState(false);
 
   useEffect(() => {
     if (sheet === 'Add') {
       const result = arrayImages.filter(item => item.id === photoId);
       setImageUrl(result[0].url);
+      setModalVisible(false);
       return navigation.navigate('Profile');
     } else if (sheet === 'Remove') {
       const result = arrayImages.filter(item => item.id !== photoId);
       setArrayImage(result);
+      setModalVisible(false);
+    } else if (sheet === 'Cancel') {
+      setModalVisible(false);
     }
```

Comment:
you have setModalVisible(false); code in all 3 cases

Reply:
Removed extra if body.